### ISSUE TITLE
x/mobile/app: android:screenOrientation is not honored

### ISSUE THREAD

**Issue by rakyll** [golang/go#10943]
I have an application with the following AndroidManifest. `android:screenOrientation="landscape"` is ignored. The application starts in the portrait mode and screen orientation changes as I rotate the device.  This value must have been handled during the EGLSurface creation and configuration changes on Android.

cc/ @crawshaw 

```
<?xml version="1.0" encoding="utf-8"?>
<manifest
  xmlns:android="http://schemas.android.com/apk/res/android"
  package="org.golang.todo.main"
  android:versionCode="1"
  android:versionName="1.0">

  <uses-sdk android:minSdkVersion="9" />
  <application android:label="Drum machine" android:hasCode="false" android:debuggable="true">
  <activity android:name="android.app.NativeActivity"
    android:label="Drums"
    android:configChanges="orientation|screenSize"
    android:screenOrientation="landscape">
    <meta-data android:name="android.app.lib_name" android:value="drummachine" />
    <intent-filter>
      <action android:name="android.intent.action.MAIN" />
      <category android:name="android.intent.category.LAUNCHER" />
    </intent-filter>
  </activity>
  </application>
</manifest>
```

**Comment by crawshaw** [comment]
I have had a change for this sitting around for a while, but I'm working on proper testing first.

**Comment by dskinner** [comment]
@crawshaw while I submitted CL 16150, I don't consider it complete (I think there's a tag the Go team uses for this but I don't recall, this is my first CL). Specifically, testing needs a definitive target and not just whatever's installed on a developers machine as in the CL.

The test should enforce use of a specific build-tool version as a method of declaring the cmd's intended output compatibility.

I considered the current test broken but there are useful bits that should remain describing the format. The comments of `skipByte` for example should be retained somehow though the actual map is no longer necessary.

As a frame of reference, I have the following build tool versions installed locally and did a byte compare of manifest output:

```
19.1.0  21.1.1  false
19.1.0  21.1.2  false
19.1.0  22.0.1  false
19.1.0  23.0.1  false
21.1.1  21.1.2  true
21.1.1  22.0.1  true
21.1.1  23.0.1  true
21.1.2  22.0.1  true
21.1.2  23.0.1 true
22.0.1  23.0.1 true
```

All resource identifiers that can not be extracted from gomobile's minimum target (api-9?) should be excluded currently. This concern is a separate issue with gomobile since resource identifiers are hard-coded but contain no information wrt the api level the identifier was introduced. This may lead to invalid manifests containing resource identifiers for an api level higher than the declared target api of a developers manifest. I'm unaware of the defined behaviour in such a case.

This should not affect `screenOrientation` but I will double check each identifier that I did add to see if they come after the minimum target, pending discussion.
